sharkfuser/include/fusili/logging.h

```diff
@@ -0,0 +1,180 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef FUSILI_LOGGING_H
+#define FUSILI_LOGGING_H
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+namespace fusili {
+
+enum class [[nodiscard]] error_code_t {
+  // Add error codes as needed
+  OK,
+  NOT_IMPLEMENTED,
+  ATTRIBUTE_NOT_SET,
+  INVALID_ATTRIBUTE,
+  TENSOR_NOT_FOUND,
+};
+
+static const std::unordered_map<error_code_t, std::string> error_code_str = {
+    {error_code_t::OK, "OK"},
+    {error_code_t::NOT_IMPLEMENTED, "NOT_IMPLEMENTED"},
+    {error_code_t::ATTRIBUTE_NOT_SET, "ATTRIBUTE_NOT_SET"},
+    {error_code_t::INVALID_ATTRIBUTE, "INVALID_ATTRIBUTE"},
+    {error_code_t::TENSOR_NOT_FOUND, "TENSOR_NOT_FOUND"}};
+
+typedef struct [[nodiscard]] error_object {
+  error_code_t code;
+  std::string err_msg;
+
+  error_object() : code(error_code_t::OK), err_msg("") {};
+  error_object(error_code_t err, std::string msg)
+      : code(err), err_msg(std::move(msg)) {};
+
+  error_code_t get_code() const { return code; }
+
+  const std::string &get_message() const { return err_msg; }
+
+  bool is_good() const { return code == error_code_t::OK; }
```

Comment:
Maybe fewer different words meaning the same thing. Just make this `is_ok()` and if you need the inverse `is_failure()`

Reply:
Done